environment/src/main/java/com/sequenceiq/environment/environment/validation/EnvironmentValidatorService.java

```diff
@@ -211,7 +213,14 @@ private void fetchSecurityGroup(EnvironmentEditDto editDto, Environment environm
     }
 
     private Set<String> getSecurityGroupIdSet(EnvironmentEditDto editDto) {
-        return Set.of(editDto.getSecurityAccess().getSecurityGroupIdForKnox(), editDto.getSecurityAccess().getDefaultSecurityGroupId());
+        Set<String> groups = new HashSet<>();
+        groups.addAll(getGroups(editDto.getSecurityAccess().getSecurityGroupIdForKnox()));
+        groups.addAll(getGroups(editDto.getSecurityAccess().getDefaultSecurityGroupId()));
+        return groups;
+    }
+
+    private Set<String> getGroups(String securityGroupIds) {
+        return getSecurityGroupIds(securityGroupIds);
     }
```

Comment:
Not a big deal, but I do not feel having a separate method justified as it simply wraps around `getSecurityGroupIds()` without taking any additional action.